[🐛 Bug]: Error Chrome driver and saucelab

### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.15.0

### Node.js Version

v18.16.0

### Mode

Standalone Mode

### Which capabilities are you using?

```typescript
{
  "browserName": 'MicrosoftEdge',
  "browserVersion": 'latest',
  "platformName": 'Windows 10',
  'sauce:options': {build: `${now}`, },
},
{
    browserName: 'chrome',
    browserVersion: '75',
    platformName: 'Windows 10',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'chrome',
    browserVersion: 'latest',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'chrome',
    browserVersion: 'latest',
    platformName: 'macOS 10.15',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'firefox',
    browserVersion: '55',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'firefox',
    browserVersion: 'latest',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'firefox',
    browserVersion: 'latest',
    platformName: 'Windows 10',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'firefox',
    browserVersion: 'latest',
    platformName: 'macOS 10.15',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'safari',
    browserVersion: '11',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'safari',
    browserVersion: '12',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'safari',
    browserVersion: '13',
    platformName: 'Mac 10.15',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'safari',
    browserVersion: '14',
    platformName: 'Mac 11.00',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'safari',
    browserVersion: 'latest',
    platformName: 'macOS 10.15',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'internet explorer',
    browserVersion: '10',
    platformName: 'Windows 2012',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'internet explorer',
    browserVersion: '11',
    platformName: 'Windows 10',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '13',
    platformName: 'Windows 10',
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '14',
    platformName: 'Windows 10',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '15',
    platformName: 'Windows 10',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '16',
    platformName: 'Windows 10',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '17',
    platformName: 'Windows 10',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '18',
    platformName: 'Windows 10',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '79',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '80',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '81',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '84',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '85',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '86',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: '87',
    platformName: 'Mac 10.13',
    'sauce:options': {build: `${now}`, },
},
{
    browserName: 'MicrosoftEdge',
    browserVersion: 'latest',
    platformName: 'macOS 10.15',
    'sauce:options': {build: `${now}`, },
},
{
    'browserName':'Safari',
    'appium:deviceName':'iPhone XS Simulator',
    'appium:platformVersion':'14.0',
    'platformName':'iOS',
    'sauce:options': {
        build: `${now}`,
        appiumVersion: '1.19.1'
    },
},
{
    'browserName':'Safari',
    'appium:deviceName':'iPhone XS Simulator',
    'appium:platformVersion':'12.0',
    'platformName':'iOS',
    'sauce:options': {
        build: `${now}`,
        appiumVersion: '1.9.1'
    },
},
{
    browserName: 'browser',
    'appium:platformVersion': '5.1',
    'appium:deviceName': 'Android Emulator',
    platformName: 'Android',
    'sauce:options': {
        build: `${now}`,
        appiumVersion: '1.18.1'
    },
},
{
    browserName: 'Chrome',
    'appium:platformVersion': '8.0',
    'appium:deviceName': 'Android Emulator',
    platformName: 'Android',
    'sauce:options': {
        build: `${now}`,
        appiumVersion: '1.18.1'
    },
},
{
    browserName: 'Chrome',
    'appium:platformVersion': '11.0',
    'appium:deviceName': 'Android GoogleAPI Emulator',
    platformName: 'Android',
    'sauce:options': {
        build: `${now}`,
        appiumVersion: '1.18.1'
    },
},
```


### What happened?

Pipeline return error.

See:

- [@borracciaBlu/dh-addeventlistener pipeline](https://github.com/borracciaBlu/dh-addeventlistener/actions/runs/5874918326/job/15930433179)
- [@borracciaBlu/dh-classlist pipeline](https://github.com/borracciaBlu/dh-classlist/actions/runs/5874917507/job/15930436109)

### What is your expected behavior?

No errors

### How to reproduce the bug.

Apparently latest chrome drivers are gone.

In one pipeline i've got `No version "79" found,` in the other `No version "80" found,`

### Relevant log output

```typescript
2023-08-16T05:14:45.118Z INFO @wdio/utils: Setting up browser binaries for: MicrosoftEdge@latest, 13, 14, 15, 16, 17, 18, 79, 80, 81, 84, 85, 86, 87 - chrome@75, latest - firefox@55, latest - internet explorer@10, 11 - browser@stable - Chrome@stable
2023-08-16T05:14:45.735Z INFO @wdio/local-runner: Shutting down spawned worker
2023-08-16T05:14:45.836Z INFO webdriver: Downloading Chromedriver v116.0.5845.96
2023-08-16T05:14:45.838Z INFO webdriver: Downloading Chromedriver v75
2023-08-16T05:14:46.254Z INFO @wdio/local-runner: Waiting for 0 to shut down gracefully
2023-08-16T05:14:46.254Z INFO @wdio/local-runner: shutting down
Error: Error: No version "79" found, latest versions available are 109.0.1518.0, 109.0.1518.0, 109.0.1518.0, 109.0.1518.0, 109.0.1518.0, 109.0.1517.0, 109.0.1517.0, 109.0.1509.0, 109.0.1509.0, 109.0.1509.0
    at fetchVersion (file:///home/runner/work/dh-classlist/dh-classlist/node_modules/edgedriver/dist/install.js:66:15)
    at async download (file:///home/runner/work/dh-classlist/dh-classlist/node_modules/edgedriver/dist/install.js:24:21)
    at async Promise.all (index 7)
    at async Promise.all (index 0)
Error: No version "79" found, latest versions available are 109.0.1518.0, 109.0.1518.0, 109.0.1518.0, 109.0.1518.0, 109.0.1518.0, 109.0.1517.0, 109.0.1517.0, 109.0.1509.0, 109.0.1509.0, 109.0.1509.0
    at fetchVersion (file:///home/runner/work/dh-classlist/dh-classlist/node_modules/edgedriver/dist/install.js:66:15)
    at async Launcher.run (file:///home/runner/work/dh-classlist/dh-classlist/node_modules/@wdio/cli/build/launcher.js:98:13)
    at async download (file:///home/runner/work/dh-classlist/dh-classlist/node_modules/edgedriver/dist/install.js:24:21)
    at async Promise.all (index 7)
    at async Promise.all (index 0)
    at async Launcher.run (file:///home/runner/work/dh-classlist/dh-classlist/node_modules/@wdio/cli/build/launcher.js:98:13)
    at async Module.run (file:///home/runner/work/dh-classlist/dh-classlist/node_modules/@wdio/cli/build/run.js:64:20)
    at async Module.run (file:///home/runner/work/dh-classlist/dh-classlist/node_modules/@wdio/cli/build/run.js:64:20)
Error: Process completed with exit code 1.
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

--- a/packages/wdio-utils/src/driver/utils.ts
+++ b/packages/wdio-utils/src/driver/utils.ts
@@ -209,16 +209,20 @@ export function setupEdgedriver (cacheDir: string, driverVersion?: string) {
 }
 
 /**
- * helper method to determine if we need to start a browser driver
- * which is: whenever the user has set connection options that differ from
- * the default, or a port is set
+ * helper method to determine if we need to setup a browser driver
+ * which is:
+ *   - whenever the user has set connection options that differ
+ *     from the default, or a port is set
+ *   - whenever the user defines `user` and `key` which later will
+ *     update the connection options
  */
-export function definesRemoteDriver(options: Pick<Options.WebDriver, 'protocol' | 'hostname' | 'port' | 'path'>) {
+export function definesRemoteDriver(options: Pick<Options.WebDriver, 'user' | 'key' | 'protocol' | 'hostname' | 'port' | 'path'>) {
     return Boolean(
         (options.protocol && options.protocol !== DEFAULT_PROTOCOL) ||
         (options.hostname && options.hostname !== DEFAULT_HOSTNAME) ||
         Boolean(options.port) ||
-        (options.path && options.path !== DEFAULT_PATH)
+        (options.path && options.path !== DEFAULT_PATH) ||
+        Boolean(options.user && options.key)
     )
 }
 

--- a/packages/wdio-utils/tests/driver/utils.test.ts
+++ b/packages/wdio-utils/tests/driver/utils.test.ts
@@ -3,7 +3,7 @@ import { describe, it, expect, vi, beforeEach } from 'vitest'
 import { getChromePath } from 'chrome-launcher'
 import { canDownload, resolveBuildId, detectBrowserPlatform } from '@puppeteer/browsers'
 
-import { parseParams, getLocalChromePath, getBuildIdByPath, setupChrome } from '../../src/driver/utils.js'
+import { parseParams, getLocalChromePath, getBuildIdByPath, setupChrome, definesRemoteDriver } from '../../src/driver/utils.js'
 
 vi.mock('chrome-launcher', () => ({
     getChromePath: vi.fn().mockReturnValue('/foo/bar')
@@ -121,5 +121,16 @@ describe('driver utils', () => {
             })
             expect(resolveBuildId).toBeCalledWith('chrome', 'windows', '1.2.3')
         })
+    })
+
+    it('definesRemoteDriver', () => {
+        expect(definesRemoteDriver({})).toBe(false)
+        expect(definesRemoteDriver({ hostname: 'foo' })).toBe(true)
+        expect(definesRemoteDriver({ port: 1 })).toBe(true)
+        expect(definesRemoteDriver({ path: 'foo' })).toBe(true)
+        expect(definesRemoteDriver({ protocol: 'foo' })).toBe(true)
+        expect(definesRemoteDriver({ user: 'foo' })).toBe(false)
+        expect(definesRemoteDriver({ key: 'foo' })).toBe(false)
+        expect(definesRemoteDriver({ user: 'foo', key: 'bar' })).toBe(true)
     })
 })

